src/commands/build-util.ts

```diff
@@ -213,3 +282,24 @@ export function generateEnvModule(mode: 'development' | 'production') {
   envObject.NODE_ENV = mode;
   return `export default ${JSON.stringify(envObject)};`;
 }
+
+/** Snowpack-specific file transformations */
+export function snowpackTransformations(
+  input: SnowpackSourceFile,
+  config: SnowpackConfig,
+): SnowpackSourceFile {
+  switch (input.baseExt) {
+    case '.js': {
+      input.code = wrapImportMeta({code: input.code, env: true, hmr: false, config});
+      return input;
+    }
+    case '.html': {
+      // replace %PUBLIC_URL% with baseUrl
+      input.code = input.code.replace(/%PUBLIC_URL%\/?/g, config.buildOptions.baseUrl);
+      return input;
+    }
+    default: {
+      return input;
+    }
+  }
+}
```

Comment:
Focus 3: centralize final transformations in one place. Like here, with `%PUBLIC_URL%`.

It should be noted that by this point, **it should only be handling browser-friendly files.** So these should only be HTML, CSS, and JS changes that Snowpack cares about here (maybe proxying, too?)